elks/arch/i86/drivers/block/bioshd.c

```diff
@@ -332,15 +332,42 @@ static int bioshd_open(struct inode *inode, struct file *filp)
     return 0;
 }
 
+void bioshd_park(struct drive_infot *drivep) {
+    unsigned int cyls = drivep->cylinders;
+    if (cyls > 1024 || drivep->fdtype != -1) return; /* int 13 AH=0xC supports only 10 bits for cyl */
+    unsigned char bios_drive = bios_drive_map[drivep - drive_info];
+    unsigned short cyl_sec = ((cyls & 0xFF) << 8) | ((cyls & 0x300) >> 2) | 0x1; /* 0x1 = sector */
+    asm (
+        "movb $0xC,%%ah\n\t"
+        "movw %0,%%cx\n\t"
+        "movb $0x0,%%dh\n\t"
+        "movb %1,%%dl\n\t"
+        "int  $0x13\n\t"
+        : /* No outputs */
+        : "r" (cyl_sec), "r" (bios_drive)
+        : "ax", "cx", "dx"
+    );
+}
+
+void bioshd_park_all(void) {
+    for (int i = 0; i < NUM_DRIVES; ++i)
+        bioshd_park(&drive_info[i]);
+}
+
 static void bioshd_release(struct inode *inode, struct file *filp)
 {
+    register struct drive_infot *drivep;
```

Comment:
`register` declarations are no longer used - GCC ignores them completely, so go ahead and delete. There are still older sections of the ELKS kernel where register declarations are seen; these cases are from old code still not changed since BCC was the compiler, which definitely required judicious use of the register keyword.